subseas_flex_fcst/layout.py

```diff
@@ -0,0 +1,393 @@
+from dash import dcc
+from dash import html
+import dash_bootstrap_components as dbc
+import dash_leaflet as dlf
+import plotly.express as px
+from widgets import Block, Sentence, Date, Units, Number
+
+import pyaconf
+import os
+
+import cptio
+from pathlib import Path
+import numpy as np
+
+CONFIG = pyaconf.load(os.environ["CONFIG"])
+DATA_PATH = CONFIG["results_path"]
+
+IRI_BLUE = "rgb(25,57,138)"
+IRI_GRAY = "rgb(113,112,116)"
+LIGHT_GRAY = "#eeeeee"
+
+# Initialization
+fcst_mu = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_mu_file"])))
```

Comment:
We shouldn't open datasets at the top level of the python file. My suggestion was to put this call to `open_cptdataset` inside `app_layout` or one of its sub-functions, so that the dataset doesn't get opened until the page is visited.

Reply:
You can set all these variables at the start of `app_layout` and then pass them into the lower-level `*_layout` functions as arguments. `app_layout` can't have arguments because it will be called by the framework, but the other functions are just internal implementation details of `app_layout` and you can change them however you want.